pallets/space-ownership/tests/src/tests.rs

```diff
@@ -5,169 +5,182 @@
+            ),
+            OwnershipError::<Test>::NotAllowedToAcceptOwnershipTransfer
         );
-    });
-}
-
-#[test]
-fn reject_pending_ownership_should_work() {
-    ExtBuilder::build_with_space().execute_with(|| {
-        assert_ok!(_transfer_default_space_ownership());
-        // Transfer SpaceId 1 owned by ACCOUNT1 to ACCOUNT2
-        assert_ok!(_reject_default_pending_ownership()); // Rejecting a transfer from ACCOUNT2
-
-        // Check whether owner was not changed
-        let space = Spaces::space_by_id(SPACE1).unwrap();
-        assert_eq!(space.owner, ACCOUNT1);
-
-        // Check whether storage state is correct
-        assert!(SpaceOwnership::pending_space_owner(SPACE1).is_none());
-    });
-}
-
-#[test]
-fn reject_pending_ownership_should_work_when_proposal_rejected_by_current_space_owner() {
-    ExtBuilder::build_with_space().execute_with(|| {
-        assert_ok!(_transfer_default_space_ownership());
-        // Transfer SpaceId 1 owned by ACCOUNT1 to ACCOUNT2
-        assert_ok!(_reject_default_pending_ownership_by_current_owner()); // Rejecting a transfer from ACCOUNT2
-
-        // Check whether owner was not changed
-        let space = Spaces::space_by_id(SPACE1).unwrap();
-        assert_eq!(space.owner, ACCOUNT1);
-
-        // Check whether storage state is correct
-        assert!(SpaceOwnership::pending_space_owner(SPACE1).is_none());
-    });
-}
 
-#[test]
-fn reject_pending_ownership_should_fail_when_space_not_found() {
-    ExtBuilder::build_with_pending_ownership_transfer_no_space().execute_with(|| {
-        assert_noop!(_reject_default_pending_ownership(), SpacesError::<Test>::SpaceNotFound);
+        assert_ok!(
+            Ownership::accept_pending_ownership(
+                RuntimeOrigin::signed(ACCOUNT2),
+                EntityWithOwnership::Space(SPACE1)
+            )
+        );
     });
 }
 
 #[test]
-fn reject_pending_ownership_should_fail_when_no_pending_transfer_on_space() {
-    ExtBuilder::build_with_space().execute_with(|| {
+fn reject_pending_ownership_should_fail_if_no_pending_transfer() {
+    ExtBuilder::build_with_all_enitities().execute_with(|| {
+        // Mock rejecting ownership transfer for a space entity with no pending transfer
         assert_noop!(
-            _reject_default_pending_ownership(),
-            SpaceOwnershipError::<Test>::NoPendingTransfer
-        ); // Rejecting a transfer from ACCOUNT2
+            Ownership::reject_pending_ownership(
+                RuntimeOrigin::signed(ACCOUNT1),
+                EntityWithOwnership::Space(SPACE1)
+            ),
+            OwnershipError::<Test>::NoPendingTransfer
+        );
     });
 }
 
 #[test]
-fn reject_pending_ownership_should_fail_when_account_is_not_allowed_to_reject() {
-    ExtBuilder::build_with_space().execute_with(|| {
-        assert_ok!(_transfer_default_space_ownership()); // Transfer SpaceId 1 owned by ACCOUNT1 to ACCOUNT2
-
+fn reject_pending_ownership_should_not_allow_non_owner_to_reject() {
```

Comment:
non owner or non target, right?

Maybe make sense to add "non_target" into test name too?